--- a/glib/gtypes.h
+++ b/glib/gtypes.h
@@ -452,7 +452,9 @@
 #define _GLIB_HAVE_STD_OVERFLOW_CHECKS
 /* https://bugzilla.gnome.org/show_bug.cgi?id=769104 */
 #elif __GNUC__ >= 5 && !defined(__INTEL_COMPILER)
+#if !(defined __clang__ && defined __arm__)
 #define _GLIB_HAVE_BUILTIN_OVERFLOW_CHECKS
+#endif
 #elif g_macro__has_builtin(__builtin_add_overflow)
 #define _GLIB_HAVE_BUILTIN_OVERFLOW_CHECKS
 #elif defined(G_OS_WIN32) && defined(HAVE_INTSAFE_H)
