diff --git a/dlls/winex11.drv/init.c b/dlls/winex11.drv/init.c
index a0bfab2fb46..57a01bbdbca 100644
--- a/dlls/winex11.drv/init.c
+++ b/dlls/winex11.drv/init.c
@@ -89,7 +89,6 @@ static X11DRV_PDEVICE *create_x11_physdev( Drawable drawable )
     physDev->gc = XCreateGC( gdi_display, drawable, 0, NULL );
     XSetGraphicsExposures( gdi_display, physDev->gc, False );
     XSetSubwindowMode( gdi_display, physDev->gc, IncludeInferiors );
-    XFlush( gdi_display );
     return physDev;
 }
 
diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index 9d25b71c992..764ccd170e3 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -220,8 +220,6 @@ void set_window_cursor( Window window, HCURSOR handle )
     }
 
     XDefineCursor( gdi_display, window, cursor );
-    /* make the change take effect immediately */
-    XFlush( gdi_display );
 }
 
 #ifdef HAVE_X11_EXTENSIONS_XINPUT2_H
@@ -377,6 +375,10 @@ static BOOL grab_clipping_window( const RECT *clip )
     if (!is_current_process_focused()) return TRUE;
 
     if (!data) return FALSE;
+
+    if (data->clipping && EqualRect( clip, &clip_rect ))
+        return TRUE;
+
     if (!(clip_window = init_clip_window())) return TRUE;
 
     if (keyboard_grabbed)
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index b033960fc26..19d20dedb35 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -2070,7 +2070,6 @@ Window create_client_window( HWND hwnd, const XVisualInfo *visual, Colormap colo
         XMapWindow( gdi_display, data->client_window );
         if (data->whole_window)
         {
-            XFlush( gdi_display ); /* make sure client_window is created for XSelectInput */
             client_window_events_enable( data, data->client_window );
         }
         TRACE( "%p xwin %lx/%lx\n", data->hwnd, data->whole_window, data->client_window );
@@ -2416,7 +2415,6 @@ LRESULT X11DRV_DesktopWindowProc( HWND hwnd, UINT msg, WPARAM wp, LPARAM lp )
         if (now > last + 5000)
         {
             XResetScreenSaver( gdi_display );
-            XFlush( gdi_display );
             last = now;
         }
         break;
@@ -2447,7 +2445,6 @@ BOOL X11DRV_CreateWindow( HWND hwnd )
         data->clip_window = XCreateWindow( data->display, root_window, 0, 0, 1, 1, 0, 0,
                                            InputOnly, default_visual.visual,
                                            CWOverrideRedirect | CWEventMask, &attr );
-        XFlush( data->display );
         NtUserSetProp( hwnd, clip_window_prop, (HANDLE)data->clip_window );
         X11DRV_DisplayDevices_RegisterEventHandlers();
     }
@@ -2791,7 +2788,6 @@ void X11DRV_SetCapture( HWND hwnd, UINT flags )
         if (!(data = get_win_data( NtUserGetAncestor( hwnd, GA_ROOT )))) return;
         if (data->whole_window)
         {
-            XFlush( gdi_display );
             XGrabPointer( data->display, data->whole_window, False,
                           PointerMotionMask | ButtonPressMask | ButtonReleaseMask,
                           GrabModeAsync, GrabModeAsync, None, None, CurrentTime );
@@ -2802,9 +2798,7 @@ void X11DRV_SetCapture( HWND hwnd, UINT flags )
     else  /* release capture */
     {
         if (!(data = get_win_data( thread_data->grab_hwnd ))) return;
-        XFlush( gdi_display );
         XUngrabPointer( data->display, CurrentTime );
-        XFlush( data->display );
         thread_data->grab_hwnd = NULL;
         release_win_data( data );
     }
